npc/other/CashShop_Functions.txt

```diff
@@ -304,3 +304,23 @@ function	script	F_Snowball	{
 	}
 	end;
 }
+
+// Status reduction potion
+//============================================================
+// - Permanently reduces base stat <type> by 1.
+// - Returns status points equals to points needed to raise
+//   that stat to original value.
+// - Doesn't work if base status <type> is already 1.
+// * callfunc("F_CashReduceStat",<type>{,<itemid>});
+function	script	F_CashReduceStat	{
+	.@type = getarg(0);
+	.@itemid = getarg(1, 0);
+	
+	if(readparam(.@type) < 2) return;
+	
+	if(.@itemid)
```

Comment:
You get the status points before and check if that status can be decreased at all. Because then the item should not be deleted. Would also be nice to add the hardcoded -1 as parameter instead, so that customs can be done that also reduce by -5 for example.